prompting-101/improving-chatGPTs-responses.md

```diff
@@ -176,6 +176,98 @@ You're likely to see a much longer response to this prompt, one which includes t
+
+![Customize Instructions](./assets/customize-chatgpt-menu.png)  
+*Fig. Account options to customize ChatGPT ([Full size image](assets/customize-chatgpt-menu.png))*
+
+- Click "OK" if you see a dialogue box introducing Custom Instructions
+
+![Custom Instructions Dialogue Box](./assets/custom-instructions-intro.png)  
+*Fig. Dialogue box introducing Custom Instructions for ChatGPT ([Full size image](assets/custom-instructions-intro.png))*
+
+- Provide details for Custom Instructions. 
+   - Add a response to the question "What do you do?". 
+      - A response could look like: "I am a full-stack software engineering student at a full-time coding program."
+   - Add a response to the question "What traits should ChatGPT have?"
+      - A response could look like: "I would like ChatGPT to provide me with pointers when I ask for help with coding problems, but I do not want ChatGPT responses to contain code or code examples because I would like to write the code solution myself. If a request could be interpreted as needing code, please explain the solution in plain language without writing any code."
+   - Add a response to the question "Anything else ChatGPT should know about you?"
+      - A response could look like: "I do not want responses from ChatGPT to provide solutions with code in them because I want to get more practice writing code by myself."
+
+
+
+![Custom Instructions Details Example](./assets/custom-instructions-without-code.png)  
+*Fig. Example of Custom Instructions for ChatGPT ([Full size image](assets/custom-instructions-without-code.png))*
+
+<details>
+   <summary>Expand this section to see an example prompt asking for help after providing Custom Instructions to not provide code in responses (from July 2025).</summary>
+
+> Please provide me a solution for the following problem: Given a string `s`, find the first non-repeating character in it and return its index. If it does not exist, return -1.
+
+> Since you asked earlier to write code yourself and only want pointers when solving problems, here’s how you can approach this one without giving away the code:
+> 
+> Problem-Solving Strategy:
+> 1. Understand the goal:
+>     * You need to find the first character in the string that appears only once.
+>     * If no such character exists, return -1.
+> 2. Break it into steps:
+>     * Traverse the string and count the frequency of each character.
+>     * Then iterate again through the string and find the first character with a count of 1.
+>     * Return the index of that character.
+> 3. Tools to use:
+>     * A dictionary or collections.Counter can help you count frequencies.
+>     * A second loop helps you check for the first character that meets the criteria.
+> 4. Edge Cases to consider:
+>     * Empty string → should return -1.
+>     * All repeating characters → should return -1.
+>     * The first character is unique → should return index 0.
+</details> 
+
```

Comment:
What if we had a final wrap-up to the coding section that has a reminder that even if we do get code results (intentionally or unintentionally) that we should still carefully review the response to make sure we understand not just the syntax of what's being suggested, but also the reason _why_ each step is in a solution?

If there are areas that are unclear, we should ask for further clarification and engage in a conversation to gain additional understanding before using any solution provided by an AI. Ideally, after reviewing the AI response, we should be able to write approximately the same approach on our own without refering back to the output from the AI, and be able to explain what the code does from both a syntax perspective, and a logcial perspective.